### DISCUSSION TITLE
How do you handle exponential backoff via reconnect on socket.io?

### DISCUSSION THREAD

**Discussion by aaronkchsu** [socketio/socket.io#4322] [Q&A] [answered]
I have been told that for the websocket client, in the socket connection retry mechanism it would be important to implement exponential backoff in the client, so as to avoid the backend from becoming overwhelmed when there are sharp new socket connections being attempted.

What settings are best to pass in with socket.io to handle this?

**Comment by aaronkchsu** [comment, marked as answer]
Edit im really slow about this sorry, 

this answers most of my questions
https://socket.io/docs/v4/client-api/#event-reconnect

**Reply by darrachequesne** [reply]
You are right! The related options are here: https://socket.io/docs/v4/client-options/#reconnection
